### PR TITLE
made world control not crash when ic2 classic is installed

### PR DESCRIPTION
**Pull request by Trinsdar**
that is all I did when it comes to ic2 classic compat. I am not doing anything else in terms of that.
I did clean up he logic in Reactor utils as it was written in a very messy way and made it harder to read.
also I updated the forge version

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -45,10 +45,10 @@ repositories {
 
 
 minecraft {
-	version = "1.12.2-14.23.1.2607"
+	version = "1.12.2-14.23.5.2778"
```
Comment (adrwa) [inline build.gradle]: Changing this stuff is out of scope for this PR

File: src/main/java/worldcontrolteam/worldcontrol/crossmod/industrialcraft2/ReactorUtils.java [hunk 2]
```diff
@@ -16,15 +16,29 @@ public static IReactor getReactorAt(World world, BlockPos pos) {
         if (world == null)
             return null;
         TileEntity tile = WCUtility.getTileEntity(world, pos).orElse(null);
-        return tile == null ? null : tile instanceof IReactor ? (IReactor) tile : tile instanceof IReactorChamber ? ((IReactorChamber) tile).getReactorInstance() : null;
+        if (tile == null){
```
Comment (adrwa) [inline src/main/java/worldcontrolteam/worldcontrol/crossmod/industrialcraft2/ReactorUtils.java]: Please maintain the same format as the original

File: src/main/java/worldcontrolteam/worldcontrol/crossmod/industrialcraft2/ReactorUtils.java [hunk 3]
```diff
@@ -16,15 +16,29 @@ public static IReactor getReactorAt(World world, BlockPos pos) {
...
+
     }
 
     public static boolean isProducing(World world, BlockPos pos) {
         return world.isBlockIndirectlyGettingPowered(pos) > 0;
     }
 
     public static int getNuclearCellTimeLeft(ItemStack rStack) {
-        return rStack.isEmpty() ? 0 : rStack.getItem() instanceof AbstractDamageableReactorComponent ? rStack.getItem() instanceof ICustomDamageItem ? ((ICustomDamageItem) rStack.getItem()).getMaxCustomDamage(rStack) - ((ICustomDamageItem) rStack.getItem()).getCustomDamage(rStack) : rStack.getMaxDamage() - rStack.getItemDamage() : 0;
+        if (rStack.isEmpty()){
```
Comment (adrwa) [inline src/main/java/worldcontrolteam/worldcontrol/crossmod/industrialcraft2/ReactorUtils.java]: Please maintain the same format as the original

File: gradle.properties [hunk 4]
```diff
@@ -1,4 +1,5 @@
-forgeVersion=1.9.4-12.17.0.1939
+forgeVersion=14.23.5.2778
+mc_version=1.12.2
```
Comment (adrwa) [inline gradle.properties]: Changing this stuff is out of scope for this PR

**Review by adrwa — CHANGES_REQUESTED** [review]

**Comment by Trinsdar** [comment]
Ok I'll change that back. The only reason I updated the forge version was because ic2 classic would crash with such old versions

File: build.gradle [hunk 5]
```diff
@@ -48,7 +48,7 @@ minecraft {
 	version = "1.12.2-14.23.1.2607"
 	runDir = "run"
 
-	mappings = "snapshot_20170919"
+	mappings = "snapshot_20171003"
```
Comment (adrwa) [inline build.gradle]: This needs changing too

**Review by adrwa — CHANGES_REQUESTED** [review]

**Review by adrwa — APPROVED** [review]

**Comment by dmf444** [comment]
Haven’t checked myself, but the API changes, does that support all types of coolers and reactor components?

Also, does IC2 classic use a different mod id?

**Comment by Trinsdar** [comment]
1. I have no idea
2. no it uses the same modid, although it also has a separate modid for checking for it specifically.

**Comment by Chocohead** [comment]
Semantically it's a little different, `IReactorComponent` covers anything that can go in a reactor (`IBaseReactorComponent` covers anything that will be allowed to stay in a reactor, but not necessarily inserted). `AbstractDamageableReactorComponent` covers everything IC2 adds that can go in a reactor that also implements `ICustomDamageItem`, ie things that wear out or get charged up. Should be fine from the way it's being used there.

**Pull request merged**